### PR TITLE
Redux store to save input from input type literal

### LINKED ISSUE #233: Using Redux Store to save input from InputLIteral
- When a user hits the submit button on the ResourceTemplateForm we can access all of the user inputted data via the state.
- https://reactjs.org/docs/lifting-state-up.html

Edit: (12/11/18)
- Update using Redux for this now, storing input in global store for InputLiteral.
- Work on Testing as well

### LINKED ISSUE #243: Undo Default Value Constraint for type Literal
This is the PR that does this: 
https://github.com/LD4P/sinopia_editor/pull/236

Type literal does not have defaultLiteral values.

### PR DESCRIPTION
**Pull request by SaravShah**
Currently, input are stored in the global redux store and are displayed when entered in the input:
screen shot 2018-12-12 at 10 55 47 am

Clicking the ❌  currently does not do anything, but working on #245

- added a commit that removes default value from value constraint in order to get redux working
closes #243
closes #233 

Let me know if you would like me to break this into separate PR's
but here is a break down of the commits:
Commit 1: Installing `redux` and `react-redux`
Commit 2: Implement Redux actions, reducers and store.
Commit 3: Adding items to the Global redux store
Commit 4:  Reformat location of actions, reducers into folders (more conventional)
Commit 5: Remove Default Literal Value #243  
Commit 6: Update inputLiteral component tests to use redux
Commit 7: Update ESLINT  See https://github.com/mysticatea/eslint-plugin-node/blob/master/docs/rules/no-unsupported-features/es-syntax.md
Commit 8: Fix Resourcetemplateform test to use new `Connect(InputLiteral)`
Commit 9: Reducers Test
Commit 10: Action Test

### CODE REVIEW

File: __tests__/components/editor/InputLiteral.test.js [hunk 1]
```diff
@@ -34,88 +33,110 @@ describe('<InputLiteral />', () => {
...
+
+describe('When the user enters input into field', ()=>{
+  let mock_wrapper;
+  // our mock formData function to replace the one provided by mapDispatchToProps
+  const mockFormDataFn = jest.fn()
+  mock_wrapper = shallow(<InputLiteral {...plProps} handleMyItemsChange={mockFormDataFn} />)
+
+  it('calls the mockFormDataFn', () => {
+    mock_wrapper.find('input').simulate("change", { target: { value: "foo" }})
+    expect(mock_wrapper.state('content_add')).toEqual('foo') /** expect state to have value onChange **/
```
Comment (jermnelson) [inline __tests__/components/editor/InputLiteral.test.js]: Nice that we can mock the state like this

**Comment by jermnelson** [comment]
LGTM except for minor request. I think @ndushay and @jgreben should review, just to see how we're using Redux.

File: __tests__/components/editor/InputLiteral.test.js (hunk 1 above)
Comment (SaravShah) [reply __tests__/components/editor/InputLiteral.test.js]: yes this is mocking the InputLiteral's local state

File: src/components/editor/InputLiteral.jsx [hunk 2]
```diff
@@ -165,4 +145,16 @@ InputLiteral.propTypes = {
   }).isRequired
 }
 
-export default InputLiteral;
\ No newline at end of file
+const mapStatetoProps = (state, props) => {
+  return {
+    formData: state.formData.find(obj => obj.id === props.propertyTemplate.propertyLabel)
```
[the remark this answers is not in the document]
Comment (SaravShah) [reply src/components/editor/InputLiteral.jsx]: happy to go either way, changing it to propertyURI would require some changes, but yeah! we can do that in another ticket?

[the remark this answers is not in the document]
Comment (SaravShah) [reply src/components/editor/InputLiteral.jsx]: Mind if make a ticket for that?

File: __tests__/components/editor/ResourceTemplateForm.test.js [hunk 3]
```diff
@@ -64,7 +64,7 @@ describe('<ResourceTemplateForm />', () => {
 
   it('renders InputLiteral nested component (b/c we have a property of type "literal")', () => {
     expect(wrapper
-      .find('div.ResourceTemplateForm InputLiteral').length)
+      .find('div.ResourceTemplateForm Connect(InputLiteral)').length)
```
Comment (jgreben) [inline __tests__/components/editor/ResourceTemplateForm.test.js]: Can you explain this to me? why is there no corresponding change to ResourceTemplateForm.jsx? Whi is this only needed din the test?

**Pull request merged**